homeassistant/components/bang_olufsen/media_player.py

```diff
@@ -336,8 +336,8 @@ async def _update_source_change(self, data: Source) -> None:
 
         # Check if source is line-in or optical and progress should be updated
         if self._source_change.id in (
-            BangOlufsenSource.LINE_IN,
-            BangOlufsenSource.SPDIF,
+            BangOlufsenSource.lineIn.name,
+            BangOlufsenSource.spdif.name,
```

Comment:
Each B&O source has an id and a friendly name, which is what the Enum was. For example this code here would fail because now the self._source_change.id would never be equal to "LINE_IN" or "Line-In" because the real source id is "lineIn", so the playback progress was never reset.

Reply:
Hmm, I am interested if we can also use translations for the state attributes as in a climate entity (for the presets). This way we could keep the friendly names out of the code. Enums should be full capital with underscores. Another option would be to make a map to translate the enum to friendly names. Sounds like a lot of duplication, but it makes it a lot cleaner.

Reply:
Ok. I've converted the enum class to just be a class with members. I think this retains the usability and readability of the enum for this use case.